### ISSUE TITLE
Prisma Client MUST provide an example of how to resolve subselections / relations

### ISSUE THREAD

**Issue by marcaaron** [prisma/prisma#3104]
As far as I can tell it seems there is no clear explanation of how to resolve subselections on related fields with the Prisma Client beta. With Prisma bindings these related fields are passed along via the info argument. In Prisma Client we must create additional type resolvers with query for the subselections by passing in the root id and chaining for the subfield. This process, since it is quite different from Prisma bindings, should IMO be more thoroughly documented. This seems to be causing confusion for people migrating from Prisma bindings to Prisma Client.

**Comment by schickling** [comment]
Great point @marcaaron. We're still working on preparing more/better examples. Does [this docs section](https://www.prisma.io/docs/prisma-client/api-reference/reading-data-JAVASCRIPT-rsc2/#using-fragments-for-fine-grained-data-access) help in the meanwhile?

**Comment by marcaaron** [comment]
Yes! The docs are very helpful to a point, but unfortunately there's no mention of the need to create resolvers for types. In the past you need only pass the info argument along but now it seems a separate resolver is needed. e.g. if you want to resolve a list of related posts on a user then you need a User resolver with a posts resolver as a subselection on a user query. 

```
User: {
  posts: async (root, args, context) => {
    const posts = await context.prisma.user({id: root.id})
    .posts()
    return posts
  }
}
```

Someone either new to Prisma client or familiar with prisma bindings alike might be utterly confused on how to resolve subselections.

**Comment by danthareja** [comment]
+1. This is one of the most confusing parts of the Prisma Client currently.

@schickling, do you have any advice on how to best handle querying relation fields for lists with the new client? Since this is not possible with the new Fluent API, and `$delegate` has been removed, what are the options?

**Comment by schickling** [comment]
Thanks a lot for your great feedback @marcaaron @danthareja! I understand how this can be confusing. We're currently working on updating the docs and adding more examples to make this more clear. Please bear with us for the moment. 🙏

**Comment by HenrySipp** [comment]
So, from what I understand, if our server uses schema delegation there is currently no solution provided by prisma-client. We are to use to continue to use prisma bindings? What exactly is the point of prisma client then, if it seems to be less feature-complete than bindings? I guess I'm confused as I thought prisma-client was the replacement to bindings and that bindings was to be deprecated.

**Comment by marcaaron** [comment]
I have since writing this noticed that there are examples in the Getting Started section of the docs which I neglected to work through having some previous experience with prisma bindings. I think there just needs to be something added to the main reference as well.

**Comment by danthareja** [comment]
Relevant thread on the forum for anyone still following this issue: https://www.prisma.io/forum/t/help-understanding-prisma-clients-value-proposition/4394/15

**Comment by sapkra** [comment]
I have a problem with types related to the `$fragments` function and the missing TypeScript documentation.
My problem is that when I'm using it like the example in the docs that the result will have the type `{}`.
I know that it is possible to do `$fragment<Type>()` but I couldn't find any generated `User` type which also has the `links` field with the `Link[]` type.

**Comment by ntziolis** [comment]
Without delegation ability the prisma client really destroys the hole graphql story. the one key reason to use prisma vs other dbs is that there was a way to pass through graphql querying info as it cust development time significantly if the client side dev can just decide to pull in more fields. One can always lock this down on the server but **not having the ability makes the new client a no-go for us** in development despite us loving the typesafety of it.

**Comment by pwgn** [comment]
With the prisma client, if you are about to fetch multiple relational fields, do you suggest to perform multiple queries like this:

```
const linksByUser = await prisma.user({ email: "alice@prisma.io" }).links()
const votesByUser = await prisma.user({ email: "alice@prisma.io" }).votes()
```
Or is `$fragments` they way forward?

In my specific case I'd like to fetch I`d like to process one of the nested objects after creation in the resolver. Similar to this:

```javascript
createItem: async (_, args, context, info) => {
  var item = context.prisma.createItem({
    name: args.input.name,
    post: {
      create: {
        number: 0
      }
    }
  }, info)

  const postId = await item.post().id()
  processPost(postId)

  return item
}
```

But when the returned `item` is awaited, it only returns the same value as `postId`. How can this be resolved with the prisma client?
